pkg/controller/certificates/signer/signer.go

```diff
@@ -305,3 +315,152 @@ func usagesToSet(usages []capi.KeyUsage) sets.String {
 	}
 	return result
 }
+
+type PCRSigningController struct {
+	pcrController       *certificates.PodCertificateRequestController
+	dynamicCertReloader dynamiccertificates.ControllerRunner
+}
+
+func NewPodClientCSRSigningController(
+	ctx context.Context,
+	client clientset.Interface,
+	pcrInformer certificatesv1alpha1informers.PodCertificateRequestInformer,
+	clock clock.Clock,
+	caFile, caKeyFile string,
+) (*PCRSigningController, error) {
+	caProvider, err := newCAProvider(caFile, caKeyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &podClientSigner{
+		caProvider: caProvider,
+		client:     client,
+		clock:      clock,
+	}
+
+	return &PCRSigningController{
+		pcrController: certificates.NewPodCertificateRequestController(
+			ctx,
+			"podcertificate-signing-pod-client",
+			client,
+			pcrInformer,
+			s.handle,
+		),
+		dynamicCertReloader: caProvider.caLoader,
+	}, nil
+}
+
+// Run the main goroutine responsible for watching and syncing jobs.
+func (c *PCRSigningController) Run(ctx context.Context, workers int) {
+	go c.dynamicCertReloader.Run(ctx, workers)
+
+	c.pcrController.Run(ctx, workers)
+}
+
+type podClientSigner struct {
+	client     clientset.Interface
+	caProvider *caProvider
+	clock      clock.Clock
+}
+
+func (s *podClientSigner) handle(ctx context.Context, req *certificatesv1alpha1.PodCertificateRequest) error {
+	// Ignore denied or failed requests
+	for _, c := range req.Status.Conditions {
+		if c.Type == certificatesv1alpha1.PodCertificateRequestConditionTypeDenied && c.Status == metav1.ConditionTrue {
+			return nil
+		}
+		if c.Type == certificatesv1alpha1.PodCertificateRequestConditionTypeFailed && c.Status == metav1.ConditionTrue {
+			return nil
+		}
+	}
+
+	// Ignore requests that have already been issued.
+	if len(req.Status.CertificateChain) > 0 {
+		return nil
+	}
+
+	if req.Spec.SignerName != certificatesv1alpha1.KubeAPIServerClientPodSignerName {
+		return nil
+	}
```

Comment:
Shouldn't these checks be first? The `len()` check for being cheap, the signer name because why would we inspect someone else's objects.

Reply:
Ditto, this was dropped from this PR.